src/AlgebraicGeometry/Schemes/AffineAlgebraicSet/Objects/Constructors.jl

```diff
@@ -36,25 +33,21 @@ Vanishing locus
 
 ```
 """
-function vanishing_locus(I::MPolyIdeal{<:MPolyElem}; check::Bool=true)
-  if check
-    Irad = radical(I)
-  else
-    Irad = I
-  end
-  X = Spec(base_ring(Irad), Irad)
-  return AffineAlgebraicSet(X, check=check)
+function vanishing_locus(I::MPolyIdeal{<:MPolyElem}; is_radical=false, check::Bool=true)
+  X = Spec(base_ring(I), I)
+  return AffineAlgebraicSet(X, is_reduced=is_radical, check=check)
 end
 
-affine_algebraic_set(I::MPolyIdeal{<:MPolyElem}; check::Bool=true) = vanishing_locus(I, check=check)
+affine_algebraic_set(I::MPolyIdeal{<:MPolyElem}; kwargs...) = vanishing_locus(I; kwargs...)
 
 @doc raw"""
-    vanishing_locus(p::MPolyRingElem, check::Bool=true)
+    vanishing_locus(p::MPolyRingElem; is_radical=false, check::Bool=true)
```

Comment:
```suggestion
    vanishing_locus(p::MPolyRingElem; is_radical::Bool=false, check::Bool=true)
```

However, the actual function `vanishing_locus` below does not seem to accept the kwarg `is_radical`.

Also, `check` remains a positional arg, when this change indicates it is now supposed to be a kwarg.